### DISCUSSION TITLE
How to convert SCSI2SD data to zuluSCSI file format

### DISCUSSION THREAD

**Discussion by tomatobeef** [ZuluSCSI/ZuluSCSI-firmware#96] [Q&A]
Hello!
I would like to know how to convert SCSI2SD data to zuluSCSI file format(HDn.hda or HDn.img).
Anyone know how to convert?

※Please excuse my poor English. From Japan.

**Comment by aperezbios** [comment]
@tomatobeef, There is no such thing as a "ZuluSCSI file format". They are simply raw files. There is no one set of instructions that would cover all cases of what you're asking for above. If you have a single SD card with a single drive volume on it, you can dump that with any raw disk utility, such as win32diskimager, dd, or any other, and then write that dumped file to a FAT-formatted SD card, with the appropriate filename, such as HD5.img

**Comment by tomatobeef** [comment]
@aperezbios 
Thank you!!
Please tell me more.
My SCSI2SD SD card is divided into four 900MB partitions(SCSI ID 0-3).
Is there a way to get this 900MB dump?

**Comment by aperezbios** [comment]
@tomatobeef check out this utility, which should help you achieve what you need to

https://github.com/dburr/scsi2sd_to_hda

**Comment by EntropyCoM** [comment]
Actually I do have the same issue. I do own 1 SCSI2SD HW (v5.2 i think) and now (!) two brand new RP2040.

My appoach was to exchange the internally installed SCSI2SD from my Kurzweil K2000 to one of the RP2040. And to my Kurzweil K2500 RS or my old AMIGA 2000.

On SCSI2SD I had copied all the Data I needed from my internal Quantum LPS240 and the SyQuest 270 + Iomega ZIP 250 from the SCSI Chain plus Data from the internal Floppy to the SCSI2SDs card. 

The Data from the SCSI2SDs SD card war EXTREMLY EASY ACCESSEBLE on my PC or MAC for backup purposes; how on earth should I transfer the Data to a cryptic HD1.hda file???
On the partition of the SD (from SCSI2SD) I could easily create directories and copy my files to it. 
That worked like charm on my K2000 after this process. 

Might be that the method of files on the SD card is a great benefit for the programming in the RP2040, but for ease of use in creating Directories and Files in this .IMG or .HDA Fies is not easy nor user-friendly.

And I do want to get copys from my AMIGA Data as well, without the hassle to deal with strange and cryptic files. 
Maybe I should sell the two zuluSCSI Cards and buy SCSI2SDs instead. To me actually a waste in time and money.

Sorry, if I look like a troll now, but I see no real chance of dealing with my problem... 
I bought the zuluSCSI as it was "advertised" as a new replacement for the SCSI2SD cards... not your fault.

But I think the ease of use in a partitioned SD Card is unbeatable. Especially for backup and later use of the files. 
And I want to use my AMIGA Files in an Amiga Emulator on my PC (I bought a Clonato EMU with licensed ROMs).

Kind regards,
EntropyCoM

**Reply by PetteriAimonen** [reply]
The configuration method is indeed a bit different, and we are trying to make it as user friendly as possible for different use cases.

Currently it is possible to use the whole SD card in raw passthrough mode, or set up one small FAT partition and several RAW partitions and configure pass-through through .ini file.

**Reply by EntropyCoM** [reply]
Thanks for the reply. :)
I am going to try this option! [RAW-Passthrough, might actually be an option that I have overseen.]
Best wishes,
EntropyCoM

**Reply by MeTalOneOEight** [reply]
There isn't really not much difference using a partition or an image file.
It really just adds one step to your workflow: mounting the image.
As a mac user you should be used to that, since most software comes on a dmg image.
Emulators work with image files all the time.
Backup of image files are even more easy than of a partition, you don't need extra software for a block copy of the partition or creating an image of it.
You see the irony?

**Reply by EntropyCoM** [reply]
@MeTalOneOEight 
You know what: The internet is a perfect place. 
Every time one posts a question or problem in a forum, some smart guy comes around that completly ingores your usecase and posts a solution that has barely something to do with your poblem. 
In the end the guy put some IRONY tags to it and leaves.

Actually I do not want to deal with .IMG Files, because they do not work on Windows on my 4 main computers - especially not out-of-the-box and I do not have the hassle with ANY kind of conversion if it would.  But it does not.
AND yes, I can create empty .IMG-Files with the solutions on this github (CMD: `fsutil file createnew HD00.hda 2000000000`), but I want to copy some of my *.KRZ Files into it, within different folders.  I have file based backup of my .*KRZ Files on my NAS. And I do not want to deal with IMaGed-/ISOed/or-DDed-Backups.

AND Yes, I do own two older MACs: one Performa 475 and one mini MAC 2012er Edition. 
And guess what, even if I create the .IMG File on a MAC, there are different kinds of IMG-Files you can create there, because they contain a type of filesystem. 

Keep in mind, that I want to use my SD-Cards with
    a) two Kurzweil K2x00 Sample/Synths     and / or
    b) with one AMIGA 2000. 
    
AND that MAC OS keeps creating stupid and useless Folders named  "__MACOSX" with additional Files for EVERY File you create on that "Drive". Those I have to deal with on AmigaOS and Kurzweils OS, too. Maybe they do not like this Folder or the contents of it.

Furthermore I do not feel to go two floors down to the room where my MACs are. On my PCs I just have to rotate my chair and see the front of my Kurzweil K 2000R V3 and K 2500RS, I can pull out the SD, put it in the PC and am done. [or should be] 

It is completely useless if I want to deal with a problem on my BMWs brakeing system, that someone suggests I should drive a Mercedes-Benz instead because it is "... much better in anyway ..." - Feel the IRONY in that?

Just my 2 cents.

**Reply by MeTalOneOEight** [reply]
__MACOSX are in Zip files created by MacOS X, you mean the ._ files. These contain meta data or file forks and are created on filesystems that do not support file forks like FAT.
On the Performa 475 with MacOS 7/8 those won't appear, too bad it doesn't support SD card readers.
The 2012 Mini has one built in but not sure how good Mojave deals with FAT16 disks like the Kurzweil uses.
That trip down two floors won't do you any good it seems beside a fitness excercise.

**Reply by PetteriAimonen** [reply]
The image files and direct partitions have different advantages, ideally we want to support both in ZuluSCSI.

Currently both work, but configuring multiple raw partitions requires manual .ini file configuration and a small FAT partition to store it. If SD card without any image files is inserted, it will show up as whole card on SCSI ID 1.

**Comment by PetteriAimonen** [comment]
If you run into any problems configuring it, just ask :)

It would be nice to have a configuration program like SCSI2SD does, but for
now direct editing of ini file is needed.
